app/src/main/java/org/schabi/newpipe/settings/UpdateSettingsFragment.java

```diff
@@ -32,7 +32,7 @@ private void checkNewVersionNow() {
         // Reset the expire time. This is necessary to check for an update immediately.
         defaultPreferences.edit()
                 .putLong(getString(R.string.update_expiry_key), 0).apply();
-        NewVersionWorker.enqueueNewVersionCheckingWork(getContext());
+        NewVersionWorker.enqueueNewVersionCheckingWork(requireContext(), true);
```

Comment:
I think you should use the MANUAL flag inside `NewVersionWorker.checkNewVersion` to determine whether to ignore the expiry time, instead of resetting the expiry time here. Also, since only one statement would be left here then, I would remove `checkNewVersionNow()` and inline the work enqueueing.